src/storage/local.rs

```diff
@@ -0,0 +1,82 @@
+use async_trait::async_trait;
+
+use tokio::sync::Mutex;
+use crate::track::Track;
+
+use super::AsyncStorage;
+
+#[derive(Default)]
+pub struct LocalStorage {
+    inner: Mutex<Vec<Track>>,
+    current_index: Mutex<u64>
+}
+
+#[async_trait]
+impl AsyncStorage for LocalStorage {
```

Comment:
I don't understand why do you return Result, is it ever Err in these functions? Or is it in case somebody else would want to implement this trait using Result?

Reply:
@YuliaProkopovych Thank you for your question! I want to future-proof my application and that is why this part has `Result<T, E>` return type.
For example, let's imagine that developer has a `Database` and wants to use `tokio_postgres` to connect to it. Developer wants to implement `AsyncStorage` for this `Database`. Methods on the connection (e.g `execute / query`) may return an error and that is why I  use a `Result<T, E>` return type in these methods.

Reply:
I was thinking about that, makes sense 👍